jekyll/_cci2/server/installation/phase-2-core-services.adoc

```diff
@@ -0,0 +1,619 @@
+
+.Installation Experience Flow Chart Phase 2
+image::server-install-flow-chart-phase2.png[Flow chart showing the installation flow for server 3.x with phase 2 highlighted]
+
+NOTE: In the following sections replace any items or credentials displayed between `< >` with your details.
+
+toc::[]
+
+CircleCI server is installed as a helm chart.
+
+[#create-a-namespace]
+== 1. Create a namespace
+Create a namespace to install the application into.
+
+[source,shell]
+----
+kubectl create ns circleci-server
+----
+
+TIP: Once you have created your namespace, we recommend setting your `kubectl` context too, with the following command: `kubectl config set-context --current --namespace <namespace>`
+
+[#pull-image-from-dockerhub]
+== 2. Pull image from DockerHub
+
+CircleCI's images are stored in a dockerhub repository. Credentials to pull the images from CircleCI's dockerhub will be provided to you as part of the onboarding process. A `docker-registry` Kubernetes secret will be used to pull images from a docker repository.
+
+* **Option 1: Your application has access to the public internet**
++
+The username and password will be provided to you.  This example pulls the images from CircleCI's repository directly.
++
+[source,shell]
+----
+kubectl create secret docker-registry regcred \
+  --docker-server=https://index.docker.io/v2/ \
+  --docker-username=<your-username> \
+  --docker-password=<your-access-token> \
+  --docker-email=<your-email>
+----
+
+* **Option 2: Your application does NOT have access to the public internet**
++
+Pull and store the images in whatever docker repository you have available.  Credentials to pull the images from CircleCI's dockerhub will be provided to you.  The `docker-registry` kubernetes secret you will use will take the following form.
++
+[source,shell]
+----
+kubectl create secret docker-registry regcred \
+  --docker-server=<your-docker-image-repo> \
+  --docker-username=<your-username> \
+  --docker-password=<your-access-token> \
+  --docker-email=<your-email>
+----
+
+[#create-helm-values]
+== 3. Create helm values
+
+Before installing CircleCI, it is recommended to create a new `values.yaml` file unique to your installation. The following describes the minimum required values to include in `values.yaml`. Additional customizations are available, see the provided `values.yaml` for all available options.
+
+For sensitive data there are two options: 
+
+* add into the values.yaml file 
+* add them as Kubernetes secrets directly
+
+This flexibility allows you to manage secrets using whichever process you prefer. 
+
+NOTE: Whichever option you choose, this sensitive information is stored as a kubernetes secret within CircleCI.
+
+[#api-token]
+=== API token
+
+The application requires a secret containing an API token. This API token is used to facilitate internal API communication to api-service. Use a random string and store it securely, CircleCI will not be able to recover this value if lost. There are two options:
```

Comment:
@lloydChris how/from where do they actually get this api token?

Reply:
Is it just a random string? is there guidance about how long it should be or anything?

Reply:
Same question for other values.yaml, Pusher secret, for example?

Reply:
For all of the random secrets, it is the customers responsibility to create and store the secret values.  In some cases we provide some guidance around length and valid characters (the [session cookie](https://github.com/circleci/circleci-docs/pull/7023/files#diff-2a3a996212c2865ca6993e7eaea9509247a93d42a9956aa09dc879cf211fd7c3R113) has to be 16 chars for example).  

The customer can create and store them however they want